scripts/common/host-packages.sh

```diff
@@ -114,9 +114,19 @@ function _dpkg_install_host_packages() {
         return 0
     fi
 
-    DEBIAN_FRONTEND=noninteractive dpkg --install --force-depends-version --force-confold "${fullpath}"/*.deb
-
-    logSuccess "Host packages ${packages[*]} installed"
+    if DEBIAN_FRONTEND=noninteractive dpkg --install --force-depends-version --force-confold "${fullpath}"/*.deb; then
+        logSuccess "Host packages ${packages[*]} installed"
+    else
+        logFail "Unable to install the packages ${packages[*]}" >&2
+        if  commandExists docker && [[ "${packages[*]}" == *"containerd.io"* ]]; then
+            # If the error was faced because of conflicts it still possible to move forward because we will migrate docker to containerd and then, uninstall docker
+            # Usually, in this case containerd deps are installed already. Therefore, we just warning but we will not stop the install.
+            logWarn "Unable to install the containerd.io package probably because an compatible version of docker package is installed.\n
+            However, it will continue and try to migrate the docker installed to containerd to perform this install.\n"
```

Comment:
@emosbaugh I changed here to address your concern.
Also, I think to address the issue (common scenario) we can leave just the impl that remove docker only when we know that k8s is not installed. It is an extra think so, I think we could remove this one. WDYT?

Reply:
If we proceed here the script will never install the expected version of containerd, correct?

im not sure about this. i do like your idea (b) but i think it is required to install the expected containerd version after migration. if that is not possible there will be no upgrade path which is a non starter.

perhaps for now we can remove this code entirely and merge to get new installations working and follow up with another pr for existing installs.

Reply:
Hi @emosbaugh,

I removed the changes in the host file to have only the uninstall for new installs (when we can check that kubectl is not installed, docker.io is installed)

Let me know wdyt? Could we try get this one merged to check it with testgrid?